File: analyze_plan.py
import pandas as pd
import json
import sys

# Ensure stdout uses utf-8
sys.stdout.reconfigure(encoding='utf-8')

try:
    xl = pd.ExcelFile("Plan.xlsx")
    result = {"sheets": xl.sheet_names, "data": {}}
    
    for sheet in xl.sheet_names:
        df = xl.parse(sheet)
        # Clean data: Replace NaN with None for valid JSON
        df = df.where(pd.notnull(df), None)
        
        result["data"][sheet] = {
            "columns": list(df.columns),
            "sample_rows": df.head(20).to_dict(orient='records'),
            "total_rows": len(df)
        }
    
    print(json.dumps(result, indent=2, default=str))
except Exception as e:
    print(f"Error: {e}")

File: analyze_plan.js
import * as XLSX from 'xlsx';
import fs from 'fs';

try {
    const buf = fs.readFileSync('Plan.xlsx');
    const wb = XLSX.read(buf);
    const result = { sheets: wb.SheetNames, data: {} };

    wb.SheetNames.forEach(sheetName => {
        const ws = wb.Sheets[sheetName];
        const data = XLSX.utils.sheet_to_json(ws, { header: 1 }); // Array of arrays
        if (data.length > 0) {
            result.data[sheetName] = {
                columns: data[0],
                sample_rows: data.slice(1, 20),
                total_rows: data.length
            };
        }
    });

    console.log(JSON.stringify(result, null, 2));
} catch (e) {
    console.error(e);
}
